test/services/GuidanceServiceSpec.scala

```diff
@@ -70,9 +72,14 @@ class GuidanceServiceSpec extends BaseSpec {
 
     "retrieve the url of the start page for the process" in new Test {
       val processId = "ext90001"
+
       MockGuidanceConnector
         .getProcess(processId)
-        .returns(Future.successful(process))
+        .returns(Future.successful(Some(process)))
+        
+      MockSessionRepository
+        .set(processId, process)
+        .returns(Future.successful(true))
```

Comment:
I guess getStartPageUrl and scratchProcess are very similar so there is little point in testing the latter. However, for completeness we should test the failure scenarios. That is
1) nothing comes back from the connector
2) the repository save fails

Reply:
as above